### PR TITLE
PR-02 Registration logic with small fixes and docstrings.

### PR DESCRIPTION
**Pull request by Shico-UA**
The "Acceptor" server in it's current state is able to receive data from the edx-platform, provides registration logic, and displays data on the world map.

### CODE REVIEW

File: acceptor/settings.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/1.10/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'vpzj&s*%wyjs7u!_g1j!5az_m3^u^kj(%p_zkayd&o@0(&w@2k'
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
+
+ALLOWED_HOSTS = [u'0.0.0.0', u'192.168.1.139', '*']
```
Comment (auraz) [inline acceptor/settings.py]: Add comment here about this values

File: acceptor/urls.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+"""acceptor URL Configuration
```
Comment (auraz) [inline acceptor/urls.py]: Follow edx doc style

File: acceptor/urls.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+The `urlpatterns` list routes URLs to views. For more information please see:
+    https://docs.djangoproject.com/en/1.10/topics/http/urls/
+Examples:
+Function views
+    1. Add an import:  from my_app import views
+    2. Add a URL to urlpatterns:  url(r'^$', views.home, name='home')
+Class-based views
```
Comment (auraz) [inline acceptor/urls.py]: Remove unused examples please

File: graph_creator/models.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from __future__ import unicode_literals
+
+from django.db import models
+
+
+class DataStorage(models.Model):
```
Comment (auraz) [inline graph_creator/models.py]: Add class doc string and module doc string everywhere please

File: graph_creator/static/graph_creator/map.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+var mymap = L.map('map', {
+    center: [30.0, -0.0],
+    zoom: 1
+    });
+
+    L.tileLayer('https://api.mapbox.com/styles/v1/mapbox/streets-v10/tiles/256/{z}/{x}/{y}?access_token=pk.eyJ1Ijoidm9sdHUiLCJhIjoiY2owbDM5ejd2MDJvNDMzcDJhdGpqYTFheiJ9.VpK2fUGFV4Vwwg1spz_pIA', {
```
Comment (auraz) [inline graph_creator/static/graph_creator/map.js]: Access token should be hardcoded?

File: graph_creator/static/graph_creator/map.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        attribution: 'Map data &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors, <a href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery © <a href="http://mapbox.com">Mapbox</a>',
+        maxZoom: 18,
+        minZoom: 3,
+        id: 'your.mapbox.project.id',
+        accessToken: 'pk.eyJ1Ijoidm9sdHUiLCJhIjoiY2owbDM5ejd2MDJvNDMzcDJhdGpqYTFheiJ9.VpK2fUGFV4Vwwg1spz_pIA'
```
Comment (auraz) [inline graph_creator/static/graph_creator/map.js]: use it as variable, as it is used in url as well.

File: graph_creator/static/graph_creator/map.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    }).addTo(mymap);
+
+    edxData.forEach(function(data){
+        L
+            .marker([data.fields.latitude, data.fields.longitude])
```
Comment (auraz) [inline graph_creator/static/graph_creator/map.js]: weird spacing, fix please

File: graph_creator/static/graph_creator/style.css [hunk 8]
```diff
@@ -0,0 +1,8 @@
+body {
+    padding: 0;
+    margin: 0;
+}
+html, body, #map {
+    height: 100%;
+    width: 100%;
+}
```
Comment (auraz) [inline graph_creator/static/graph_creator/style.css]: Add new line at the end of file

File: graph_creator/tests.py [hunk 9]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (auraz) [inline graph_creator/tests.py]: Think about which tests you should have here.

File: graph_creator/views.py [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+        students_amount = received_data.get('students_amount')
+        latitude = received_data.get('latitude')
+        longitude = received_data.get('longitude')
+        platform_url = received_data.get('platform_url')
+        secret_token = received_data.get('secret_token')
+
+        if secret_token is None:
+            secret_token = uuid.uuid4().hex
+            obj = DataStorage.objects.create(**kwargs)
+            obj.secret_token = secret_token
```
Comment (auraz) [inline graph_creator/views.py]: move it to the create call please, and platform url as well

File: graph_creator/views.py [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+            obj.platform_url = platform_url
+            obj.save()
+            reverse_token = requests.post('http://192.168.1.139:8000/acceptor_data/',
```
Comment (auraz) [inline graph_creator/views.py]: use edx python code style plz

File: graph_creator/views.py [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+                                          data={"reverse_token": secret_token})
+        else:
+            obj = DataStorage.objects.get(secret_token=str(secret_token))
+            obj.courses_amount = int(courses_amount)
```
Comment (auraz) [inline graph_creator/views.py]: can we use .update ?

File: acceptor/settings.py (hunk 1 above)
Comment (Shico-UA) [inline acceptor/settings.py]: Done

File: acceptor/urls.py (hunk 2 above)
Comment (Shico-UA) [inline acceptor/urls.py]: Done

File: acceptor/urls.py (hunk 3 above)
Comment (Shico-UA) [inline acceptor/urls.py]: Done

File: graph_creator/models.py (hunk 4 above)
Comment (Shico-UA) [inline graph_creator/models.py]: Done

File: graph_creator/static/graph_creator/map.js (hunk 5 above)
Comment (Shico-UA) [inline graph_creator/static/graph_creator/map.js]: Assigned it to separate variable.

File: graph_creator/static/graph_creator/map.js (hunk 6 above)
Comment (Shico-UA) [inline graph_creator/static/graph_creator/map.js]: Done

File: graph_creator/static/graph_creator/map.js (hunk 7 above)
Comment (Shico-UA) [inline graph_creator/static/graph_creator/map.js]: Changed

File: graph_creator/static/graph_creator/style.css (hunk 8 above)
Comment (Shico-UA) [inline graph_creator/static/graph_creator/style.css]: Done

File: graph_creator/tests.py (hunk 9 above)
Comment (Shico-UA) [inline graph_creator/tests.py]: Ok, I'll think about on the next iteration when learn how to write tests.

File: graph_creator/views.py (hunk 10 above)
Comment (Shico-UA) [inline graph_creator/views.py]: Done

File: graph_creator/views.py (hunk 11 above)
Comment (Shico-UA) [inline graph_creator/views.py]: Done

File: graph_creator/views.py (hunk 12 above)
Comment (Shico-UA) [inline graph_creator/views.py]: Done

File: acceptor/settings.py [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/1.10/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'vpzj&s*%wyjs7u!_g1j!5az_m3^u^kj(%p_zkayd&o@0(&w@2k'
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
```
Comment (auraz) [inline acceptor/settings.py]: Set to False

File: acceptor/settings.py [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+
+# Settled for all hosts for testing purposes.
+ALLOWED_HOSTS = ['*']
```
Comment (auraz) [inline acceptor/settings.py]: Start asking in company channel for domain name for this tool

File: graph_creator/tests.py (hunk 9 above)
Comment (auraz) [inline graph_creator/tests.py]: OK

File: graph_creator/views.py [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+    pass serialized data to the template. The template displays a map of the world with the
+    edx-platform marker on it.
+    
+    `edx_data_as_json` is a `DataStorage` containing all information about edx-platform.
+    
+    Returns http response and passes `edx_data_as_json` data as a context variable `edx_data`
+    to the `index.html` template.
+    """
+
+    def get(self, request, *args, **kwargs):
```
Comment (auraz) [inline graph_creator/views.py]: Add docstring

**Comment by auraz** [comment]
Please add description for pull request.
Couple of small comments, please fix them.
Other looks good. Please merge after addressing. 👍

File: acceptor/settings.py (hunk 13 above)
Comment (Shico-UA) [inline acceptor/settings.py]: rDone, also added private settings file.

File: acceptor/settings.py (hunk 14 above)
Comment (Shico-UA) [inline acceptor/settings.py]: Ok' I'll ask.

File: graph_creator/views.py (hunk 15 above)
Comment (Shico-UA) [inline graph_creator/views.py]: Done

**Pull request merged**